src/audio/src/src_ipc3.c

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2017 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include <sof/common.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/audio/ipc-config.h>
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/src/src.h>
+#include <sof/audio/src/src_config.h>
+#include <sof/audio/sink_api.h>
+#include <sof/audio/source_api.h>
+#include <sof/audio/sink_source_utils.h>
+#include <rtos/panic.h>
+#include <sof/ipc/msg.h>
+#include <rtos/alloc.h>
+#include <rtos/init.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/math/numbers.h>
+#include <sof/platform.h>
+#include <rtos/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <ipc4/base-config.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <limits.h>
+
+LOG_MODULE_DECLARE(src, CONFIG_SOF_LOG_LEVEL);
+
+int src_rate_check(const void *spec)
+{
+	const struct ipc_config_src *ipc_src = spec;
+
+	if (ipc_src->source_rate == 0 && ipc_src->sink_rate == 0)
+		return -EINVAL;
+
+	return 0;
+}
+
+int src_stream_pcm_sink_rate_check(struct ipc_config_src cfg,
+				   struct sof_ipc_stream_params *params)
+{
+	/* In playback, module adapter mod->stream_params from prepare() is for sink side */
+	if (cfg.sink_rate && params->rate != cfg.sink_rate)
+		return -EINVAL;
+
+	return 0;
+}
+
+int src_stream_pcm_source_rate_check(struct ipc_config_src cfg,
+				     struct sof_ipc_stream_params *params)
+{
+	/* In capture, module adapter mod->stream_params from prepare() is for source side */
+	if (cfg.source_rate && params->rate != cfg.source_rate)
+		return -EINVAL;
+
+	return 0;
+}
+
+int src_set_params(struct processing_module *mod, struct sof_sink __sparse_cache *sink)
+{
+	return 0;
+}
+
+void src_set_sink_params(struct comp_dev *dev, struct sof_source __sparse_cache *source,
+			 struct sof_sink __sparse_cache *sink)
+{
```

Comment:
well, this function now isn't really setting sink parameters? Please choose a different name

Reply:
@singalsu any suggestions for appropriate name.

Reply:
In IPC4 build the same name function sets source and sink parameters and these component data members are updated in init() where directly received from kernel.

If IPC3 function name is changed, then it could be src_get_source_sink_params(), since in IPC3 these are retrieved from params walk result for buffers.